src/System.Collections/src/System/Collections/Generic/CollectionExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace System.Collections.Generic
+{
+    public static class CollectionExtensions
+    {
+        public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key)
+        {
+            if (dictionary == null)
+            {
+                throw new ArgumentNullException(nameof(dictionary));
+            }
+
+            TValue value;
+            dictionary.TryGetValue(key, out value);
+            return value;
+        }
+
+        public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key, TValue defaultValue)
+        {
+            if (dictionary == null)
+            {
+                throw new ArgumentNullException(nameof(dictionary));
+            }
+
+            TValue value;
+            if (dictionary.TryGetValue(key, out value))
+                return value;
+            return defaultValue;
+        }
+
+        public static TValue GetValueOrDefault<TKey, TValue>(this IReadOnlyDictionary<TKey, TValue> dictionary, TKey key)
+        {
+            if (dictionary == null)
+            {
+                throw new ArgumentNullException(nameof(dictionary));
+            }
+
+            TValue value;
+            dictionary.TryGetValue(key, out value);
+            return value;
+        }
```

Comment:
This can be implemented in terms of the other overload to prevent code duplication. Same for the IDictionary method.